tree-sitter-highlight.el

```diff
@@ -0,0 +1,314 @@
+;;; tree-sitter-highlight.el --- Highlighting of buffers -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2019  Timo von Hartz
+;;
+;; Author: Timo von Hartz <c0untlizzi@gmail.com>
+
+;;; Commentary:
+
+;; Very early implementation of highlighting.
+;;; Code:
+
+(require 'tree-sitter)
+
+(require 'f)
```

Comment:
This dependency should be declared in `Cask`.

However, I think `tree-sitter` is fundamental/low-level enough that fewer external dependencies are desirable. If we just want 1-2 functions from `f`, we can reimplement them.

Reply:
I've added it to `Cask` for now, I'm just using `f-join` currently, we can re-implement that before we merge this. It's probably not needed if we bundle all the `highlights.scm` somewhere.